spdx/writers/jsonyamlxml.py

```diff
@@ -383,11 +386,18 @@ def create_snippet_info(self):
             if snippet.has_optional_field("license_comment"):
                 snippet_object["licenseComments"] = snippet.license_comment
 
+            if snippet.line_range[0] is not None:
```

Comment:
I missed this in the typing before (sorry!), but I think the range types should be `Optional[Tuple[int, int]]`, and they should be initialized with `None`. It is done the same way for other mandatory and optional fields. Then you should be able to use `has_optional_field` here

Reply:
fixed